### PR TITLE
Closes #117 117_expect_names: add expect_names argument to assert_vars()

### LINKED ISSUE #117: Feature Request: Add `expect_names` argument to `assert_vars()`
### Feature Idea

The `expect_names` argument should be added to `assert_vars()`. If set to `TRUE`, it is checked if names are provided for all variables.

This is required for https://github.com/pharmaverse/admiral/issues/1477.

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by bundfussr**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page.
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging.
- [x] Address all merge conflicts and resolve appropriately.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: tests/testthat/test-assertions.R [hunk 1]
```diff
@@ -1,7 +1,8 @@
 library(admiral.test)
```
Comment (bms63) [inline tests/testthat/test-assertions.R:1]: Can you remove the library call

File: tests/testthat/test-assertions.R [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
 
-test_that("Test 1 : `assert_has_variables` an error is thrown if a required
-           variable is missing", {
+# assert_has_variables ----
+## Test 1: error if a required variable is missing ----
+test_that("assert_has_variables Test 1: error if a required variable is missing", {
   data(admiral_dm)
```
Comment (bms63) [inline tests/testthat/test-assertions.R:6]: remove library call and replace with `admiral.test::admiral_dm`

File: R/assertions.R [hunk 3]
```diff
@@ -437,9 +437,15 @@ assert_filter_cond <- function(arg, optional = FALSE) {
 #' Checks if an argument is a valid list of variables created using `vars()`
 #'
 #' @param arg A function argument to be checked
+#'
 #' @param optional Is the checked parameter optional? If set to `FALSE` and `arg`
 #' is `NULL` then an error is thrown
 #'
+#' @param expect_names Expect Names?
```
Comment (bms63) [inline R/assertions.R:444]: The Expect Names? throws me off a little bit.  Why have it like that?

File: tests/testthat/test-assertions.R [hunk 4]
```diff
@@ -10,14 +11,16 @@ test_that("Test 1 : `assert_has_variables` an error is thrown if a required
   )
 })
 
-test_that("Test 2 : `assert_has_variables` no error is thrown if a required
-           variable exists", {
+## Test 2: no error if a required variable exists ----
+test_that("assert_has_variables Test 2: no error if a required variable exists", {
   data(admiral_dm)
```
[the remark this answers is not in the document]
Comment (bms63) [reply tests/testthat/test-assertions.R:16]: if you see any others in this test file can you please update?

**Comment by bms63** [comment]
Do you mind fixing the link issue in this PR.  I'm not sure this is best approach, but could devel get added to the two below:

https://pharmaverse.github.io/admiraldev/devel/reference/
https://pharmaverse.github.io/admiraldev/devel/articles/

image

File: tests/testthat/test-assertions.R (hunk 1 above)
Comment (bundfussr) [reply tests/testthat/test-assertions.R]: Removed

File: tests/testthat/test-assertions.R (hunk 2 above)
Comment (bundfussr) [reply tests/testthat/test-assertions.R]: I have created a dataset using `tribble()`. I think this is clearer and faster.

File: tests/testthat/test-assertions.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (bundfussr) [reply tests/testthat/test-assertions.R]: I have removed all admiral.test dependencies in the file.

File: R/assertions.R (hunk 3 above)
Comment (bundfussr) [reply R/assertions.R]: I have remove the short description ("Expect Names?").

**Review by bms63 — APPROVED** [review]
Thanks Stefan!

**Pull request merged**